modules/uploadDataReqTree.R

```diff
@@ -0,0 +1,65 @@
+# This is the module to
+# , width = "100%", height = "100%"
+uploadDataReqTreeUI <- function(id, width = "100%", height = "400px") {
+
+  # namespace
+  ns <- NS(id)
+  tagList(
+    d3Output(ns("tree"), width = width, height = height),
+    helpText(
+      tags$i(style = "color: grey; margin-right: 5px", icon("circle"), " Project"),
+      tags$i(style = "color: #694489; margin-right: 5px", icon("circle"), "Complete"),
+      tags$i(style = "color: #F7DC6F; margin-right: 5px", icon("circle"), "Not Complete"),
+      tags$i(style = "color: #28a745; margin-right: 5px", icon("circle"), "Uploaded"),
+      tags$i(style = "color: #E53935;", icon("circle"), " Not Uploaded"),
+      br(),
+      "Click nodes to expand the required manifests"
+    )
+  )
+}
+
+uploadDataReqTreeServer <- function(id, upload_data, upload_require_data, project_name) {
+  moduleServer(
+    id,
+    function(input, output, session) {
+      output$tree <- renderD3({
+        upData <- isolate(upload_data)
+        reqData <- isolate(upload_require_data)
+
+        if (length(upData) == 0) {
+          tree_list <- NULL
+        } else {
+
+          # remove project name in children to trim long names
+          pattern <- paste0(str_replace(project_name, " ", "_"), "_")
+          n_file <- length(upData$folder)
+          file_has_miss <- reqData[, c("folder", "has_miss")] %>%
```

Comment:
`has_miss`: dataset contains any missing requirement children. That will be used to determine the color of nodes (orange or purple)